src/nnet3/nnet-simple-component.h

```diff
@@ -1643,6 +1643,109 @@ class MaxpoolingComponent: public Component {
 };
 
 /**
+ * Maxpooling3dComponent implements 3d-maxpooling.
+ * This component is designed to be used after a ConvolutionComponent
+ * so that the input matrix is propagated from a 2d/3d convolutional layer.
```

Comment:
there is only one type of convolution layer, this does pseudo 3d convolution (i.e., there is no hopping along the 3rd axis).